README.md

```diff
@@ -915,178 +886,231 @@ You can replace the embedded model with a new model like this:
 
-### MongoDB specific operations
+If you want this functionality to work both ways, your SQL-models will need to use the `Jenssegers\Mongodb\Eloquent\HybridRelations` trait.
 
-**Cursor timeout**
+**This functionality only works for `hasOne`, `hasMany` and `belongsTo`.**
 
-To prevent MongoCursorTimeout exceptions, you can manually set a timeout value that will be applied to the cursor:
+The MySQL model should use the `HybridRelations` trait:
 
 ```php
-DB::collection('users')->timeout(-1)->get();
-```
+use Jenssegers\Mongodb\Eloquent\HybridRelations;
 
-**Upsert**
+class User extends Model
+{
+    use HybridRelations;
 
-Update or insert a document. Additional options for the update method are passed directly to the native update method.
+    protected $connection = 'mysql';
 
-```php
-DB::collection('users')->where('name', 'John')
-                       ->update($data, ['upsert' => true]);
+    public function messages()
+    {
+        return $this->hasMany(Message::class);
+    }
+}
 ```
+Within your MongoDB model, you should define the relationship:
 
-**Projections**
+```php
+use Jenssegers\Mongodb\Eloquent\Model;
 
-You can apply projections to your queries using the `project` method.
+class Message extends Model
+{
+    protected $connection = 'mongodb';
 
-```php
-DB::collection('items')->project(['tags' => ['$slice' => 1]])->get();
-DB::collection('items')->project(['tags' => ['$slice' => [3, 7]]])->get();
+    public function user()
+    {
+        return $this->belongsTo(User::class);
+    }
+}
 ```
 
-**Projections with Pagination**
+### Authentication
+If you want to use Laravel's native Auth functionality, register this included service provider:
 
 ```php
-$limit = 25;
-$projections = ['id', 'name'];
-DB::collection('items')->paginate($limit, $projections);
+Jenssegers\Mongodb\Auth\PasswordResetServiceProvider::class,
 ```
 
+This service provider will slightly modify the internal DatabaseReminderRepository to add support for MongoDB based password reminders.
 
-**Push**
+If you don't use password reminders, you don't have to register this service provider and everything else should work just fine.
 
-Add items to an array.
+### Queues
+If you want to use MongoDB as your database backend, change the driver in `config/queue.php`:
 
 ```php
-DB::collection('users')->where('name', 'John')->push('items', 'boots');
-DB::collection('users')->where('name', 'John')->push('messages', ['from' => 'Jane Doe', 'message' => 'Hi John']);
+'connections' => [
+    'database' => [
+        'driver' => 'mongodb',
+        'table' => 'jobs',
+        'queue' => 'default',
+        'expire' => 60,
+    ],
+],
 ```
 
-If you don't want duplicate items, set the third parameter to `true`:
+If you want to use MongoDB to handle failed jobs, change the database in `config/queue.php`:
 
 ```php
-DB::collection('users')->where('name', 'John')->push('items', 'boots', true);
+'failed' => [
+    'driver' => env('QUEUE_FAILED_DRIVER', 'database'),
```

Comment:
What is the profit to determine new env variable  `QUEUE_FAILED_DRIVER` instead `QUEUE_DRIVER`

Reply:
Ok, you copy from laravel implementation https://github.com/laravel/laravel/blob/master/config/queue.php#L84